EventListener/TokenSubscriber.php

```diff
@@ -323,4 +333,138 @@ private function getCustomFieldValues(CustomObject $customObject, Token $token,
 
         return $fieldValues;
     }
+
+    public function onTokenReplacement(TokenReplacementEvent $event): void
+    {
+        $clickthrough = $event->getClickthrough();
+
+        if (!array_key_exists('dynamicContent', $clickthrough)) {
+            return;
+        }
+
+        $lead      = $event->getLead();
+        $tokenData = $clickthrough['dynamicContent'];
+
+        foreach ($tokenData as $data) {
+            $filterContent = $data['content'];
+
+            $isCustomObject = false;
+            foreach ($data['filters'] as $filter) {
+                $customFieldValues = $this->getCustomFieldDataForLead($filter['filters'], (int) $lead['id']);
+
+                if (!empty($customFieldValues)) {
+                    $isCustomObject = true;
+                    $lead           = array_merge($lead, $customFieldValues);
+                }
+
+                if ($isCustomObject && $this->matchFilterForLead($filter['filters'], $lead)) {
+                    $filterContent = $filter['content'];
+                    break;
+                }
+            }
+
+            if ($isCustomObject) {
+                $event->addToken('{dynamiccontent="'.$data['tokenName'].'"}', $filterContent);
+            }
+        }
+    }
+
+    /**
+     * @param array<mixed> $filters
+     *
+     * @return array<mixed>
+     */
+    private function getCustomFieldDataForLead(array $filters, int $leadId): array
+    {
+        $customFieldValues = $cachedCustomItems = [];
+
+        foreach ($filters as $condition) {
+            try {
+                if ('custom_object' !== $condition['object']) {
+                    continue;
+                }
+
+                if ('cmf_' === substr($condition['field'], 0, 4)) {
+                    $customField  = $this->customFieldModel->fetchEntity(
+                        (int) explode('cmf_', $condition['field'])[1]
+                    );
+                    $customObject = $customField->getCustomObject();
+                    $fieldAlias   = $customField->getAlias();
+                } elseif ('cmo_' === substr($condition['field'], 0, 4)) {
+                    $customObject = $this->customObjectModel->fetchEntity(
+                        (int) explode('cmo_', $condition['field'])[1]
+                    );
+                    $fieldAlias   = 'name';
+                } else {
+                    continue;
+                }
+
+                $key = $customObject->getAlias().'-'.$leadId;
```

Comment:
```suggestion
                $key = $customObject->getId().'-'.$leadId;
```
Let's use ID as it is safer.